config/plop/component/component.generator.js

```diff
@@ -13,34 +13,84 @@ module.exports = {
       name: 'styledComponents',
       message: 'Do you want the component to be styled with Styled Components?',
       default: true
+    },
+    {
+      type: 'list',
+      name: 'componentType',
+      message: 'Select component type',
+      choices: ['class', 'function']
     }
   ],
-  actions: (answers) => {
+  actions: answers => {
     const actions = []
 
-    if (!answers.styledComponents) {
-      actions.push({
-        type: 'add',
-        path: `${paths.components}/{{properCase componentName}}/{{properCase componentName}}.js`,
-        templateFile: './config/plop/component/component.js.plop'
-      })
-    } else {
-      actions.push({
+    const componentTemplates = {
+      defaultClass: {
         type: 'add',
-        path: `${paths.components}/{{properCase componentName}}/{{properCase componentName}}.js`,
-        templateFile: './config/plop/component/component.styled.js.plop'
-      }, {
+        path: `${
+          paths.components
+        }/{{properCase componentName}}/{{properCase componentName}}.js`,
+        templateFile: './config/plop/component/class/component.js.plop'
+      },
+      defaultFunction: {
         type: 'add',
-        path: `${paths.components}/{{properCase componentName}}/styles.js`,
-        templateFile: './config/plop/component/styles.js.plop'
-      }, {
-        type: 'addComponentToIndex'
-      })
+        path: `${
+          paths.components
+        }/{{properCase componentName}}/{{properCase componentName}}.js`,
+        templateFile: './config/plop/component/function/component.js.plop'
+      },
+      styledClass: [
+        {
```

Comment:
I'd like to keep the object array whitespace the same. I know it's personal preference, but I don't think standardjs is opinionated on this.

Reply:
Sure, is there some way to change this in the Standard formatter extension? This is the default behavior:
![](https://i.gyazo.com/a2e7efcbe50e800f9a7d45f49150cb7f.mp4)

Reply:
Hmmm, I don't use this extension. I just rely on ESLint. I wonder if they have slightly different rulesets or what. Anyways, I'll look into that later as it's not super important.